python/python/bystro/ancestry/ancestry_types.py

```diff
@@ -153,73 +82,60 @@ class SuperpopVector:
 
     """
 
-    AFR: ProbabilityInterval = field(validator=ProbIntValidator)
-    AMR: ProbabilityInterval = field(validator=ProbIntValidator)
-    EAS: ProbabilityInterval = field(validator=ProbIntValidator)
-    EUR: ProbabilityInterval = field(validator=ProbIntValidator)
-    SAS: ProbabilityInterval = field(validator=ProbIntValidator)
+    AFR: ProbabilityInterval
+    AMR: ProbabilityInterval
+    EAS: ProbabilityInterval
+    EUR: ProbabilityInterval
+    SAS: ProbabilityInterval
+
 
-@attrs.frozen(kw_only=True)
-class AncestryTopHit:
+class AncestryTopHit(Struct):
     """
     The top hit for a sample, with the max value (a probability) and the population(s) corresponding
     """
 
-    probability: float = field(validator=is_python_unit_float)
-    populations: list[str] = field(validator=instance_of(list))
+    probability: float
+    populations: list[str]
 
-@attrs.frozen(kw_only=True)
-class AncestryResult:
-    """An ancestry result from a sample.
+    def __post_init__(self):
+        if not isinstance(self.probability, float):
+            raise TypeError(f"probability must be a float, not {type(self.probability)}")
+
+        if self.probability < LOWER_UNIT_BOUND or self.probability > UPPER_UNIT_BOUND:
+            raise TypeError(f"probability must be between {LOWER_UNIT_BOUND} and {UPPER_UNIT_BOUND}")
+
+
+class AncestryScoresOneSample(Struct, frozen=True):
+    """An ancestry result for a sample.
 
     Represents ancestry model output for an individual study
     participant (identified by sample_id) with estimates for
     populations and superpopulations, and the overall fraction of
     expected variants found missing in the sample.
     """
 
-    sample_id: str = field(validator=instance_of(str))
-    top_hit: AncestryTopHit = field(validator=instance_of(AncestryTopHit))
-    populations: PopulationVector = field(validator=instance_of(PopulationVector))
-    superpops: SuperpopVector = field(validator=instance_of(SuperpopVector))
-    # needs to be literal float for msgspec
-    missingness: float = field(converter=float, validator=unit_float_validator)
+    sample_id: str
+    top_hit: AncestryTopHit
+    populations: PopulationVector
+    superpops: SuperpopVector
+    missingness: float
 
+    def __post_init__(self):
+        # Due to PEP 484, typescheckers do not distinguish between float and int
```

Comment:
I think this is a small typo?
```suggestion
        # Due to PEP 484, type checkers do not distinguish between float and int
```